lib/async_hooks.js

```diff
@@ -461,6 +461,21 @@ function init(asyncId, type, triggerId, resource) {
   processing_hook = false;
 }
 
+const JSSideProviders = {
+  Timeout: 'timers Timeout',
+  Immediate: 'timers Immediate',
+  TickObject: 'next_tick TickObject'
+};
+
+function _getTypeName(module, type) {
+  return Object.keys(JSSideProviders)
+    .find((k) => JSSideProviders[k] === `${module} ${type}`);
+}
```

Comment:
Yeah definatly a kludge.
This is what I came up with in my current (sleep deprived) state :anguished:

I'm looking for a structure that:
1. Can get the `keys` out easily
2. can be queried for a specific name by modules and type.